samples/excel/85-preview-apis/data-types-formatted-number.yaml

```
@@ -0,0 +1,160 @@
+
+            // Get the Sample worksheet and a range on that sheet.
+            const sheet = context.workbook.worksheets.getItemOrNullObject("Sample");
+            const dateRange = sheet.getRange("A1");
+
+            // Write a number formatted as a date to cell A1.
+            dateRange.valuesAsJson = [
+              [
+                {
+                  type: Excel.CellValueType.formattedNumber,
+                  basicValue: 32889.0,
+                  numberFormat: "m/d/yyyy"
+                }
+              ]
+            ];
+            await context.sync();
+          });
+        }
+
+        async function setFormattedNumberCurrency() {
+          await Excel.run(async (context) => {
+            // This method creates a formatted number data type,
+            // and sets the format of this data type as a currency.
+
+            // Get the Sample worksheet and a range on that sheet.
+            const sheet = context.workbook.worksheets.getItemOrNullObject("Sample");
+            const currencyRange = sheet.getRange("A2");
+
+            // Write a number formatted as currency to cell A2.
+            currencyRange.valuesAsJson = [
+              [
+                {
+                  type: Excel.CellValueType.formattedNumber,
+                  basicValue: 12.34,
+                  numberFormat: "$* #,##0.00"
+                }
+              ]
+            ];
+
+            await context.sync();
+          });
+        }
+
+        async function getFormattedNumber() {
+          await Excel.run(async (context) => {
+            // This method prints information about data types 
+            // in cells A1 and A2 to the console.
+
+            // Get the Sample worksheet and two ranges on that sheet.
+            const sheet = context.workbook.worksheets.getItemOrNullObject("Sample");
+            const currencyRange = sheet.getRange("A2");
+            const dateRange = sheet.getRange("A1");
+
+            // Load the data type property of the ranges.
+            currencyRange.load("valuesAsJson");
+            dateRange.load("valuesAsJson");
+            await context.sync();
+
+            const currencyValues = currencyRange.valuesAsJson[0][0];
+            const dateValues = dateRange.valuesAsJson[0][0];
+
+            // Print information about the data types to the console.
+            console.log("Date");
+            console.log("  Type: " + dateValues.type);
+            console.log("  Basic value: " + dateValues.basicValue);
+            console.log("  Basic type: " + dateValues.basicType);
+            console.log("  Number format: " + dateValues.numberFormat);
+
+            console.log("Currency");
+            console.log("  Type: " + currencyValues.type);
+            console.log("  Basic value: " + currencyValues.basicValue);
+            console.log("  Basic type: " + currencyValues.basicType);
+            console.log("  Number format: " + currencyValues.numberFormat);
```

Same comment as above.
```suggestion
            console.log("  Number format: " + (currencyValues as Excel.FormattedNumberCellValue).numberFormat);
```